frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/validation/PhoneNumberCodeProcessor.java

```diff
@@ -99,29 +138,49 @@ public Optional<ErrorResponse> validateCode() {
 
     @Override
     public void processSuccessfulCodeRequest(String ipAddress, String persistentSessionId) {
-        switch (codeRequest.getJourneyType()) {
-            case REGISTRATION:
-                dynamoService.updatePhoneNumberAndAccountVerifiedStatus(
-                        emailAddress, codeRequest.getProfileInformation(), true, true);
-                submitAuditEvent(
-                        FrontendAuditableEvent.UPDATE_PROFILE_PHONE_NUMBER,
-                        MFAMethodType.SMS,
-                        codeRequest.getProfileInformation(),
-                        ipAddress,
-                        persistentSessionId,
-                        false);
-                break;
-            case ACCOUNT_RECOVERY:
-                dynamoService.setVerifiedPhoneNumberAndRemoveAuthAppIfPresent(
-                        emailAddress, codeRequest.getProfileInformation());
-                submitAuditEvent(
-                        FrontendAuditableEvent.UPDATE_PROFILE_PHONE_NUMBER,
-                        MFAMethodType.SMS,
-                        codeRequest.getProfileInformation(),
-                        ipAddress,
-                        persistentSessionId,
-                        true);
-                break;
+        if (codeRequest.getJourneyType() == JourneyType.REGISTRATION
+                || codeRequest.getJourneyType() == JourneyType.ACCOUNT_RECOVERY) {
+            JourneyType journeyType = codeRequest.getJourneyType();
+            UserProfile userProfile = userContext.getUserProfile().get();
+            boolean phoneNumberVerified = userProfile.isPhoneNumberVerified();
+            String phoneNumber =
+                    PhoneNumberHelper.formatPhoneNumber(codeRequest.getProfileInformation());
+            boolean updatedPhoneNumber = !phoneNumber.equals(userProfile.getPhoneNumber());
+
+            if (configurationService.isPhoneCheckerWithReplyEnabled()
+                    && (journeyType != JourneyType.ACCOUNT_RECOVERY || updatedPhoneNumber)) {
+                Session session = userContext.getSession();
+                String internalCommonSubjectIdentifier =
+                        session != null ? session.getInternalCommonSubjectIdentifier() : "";
+
+                var phoneNumberRequest =
+                        new PhoneNumberRequest(
+                                phoneNumberVerified,
+                                phoneNumber,
+                                updatedPhoneNumber,
+                                journeyType,
+                                internalCommonSubjectIdentifier);
+                try {
+                    sqsClient.send(objectMapper.writeValueAsString(phoneNumberRequest));
+                } catch (Exception e) {
+                    LOG.error("Unexpected exception: {}", e.getMessage());
```

Comment:
I wonder about putting in a more useful error message here, like "Unexpected exception when writing phone number request to SQS" or equivalent?